main.go

```diff
@@ -95,9 +103,20 @@ func main() {
 			os.Exit(1)
 		}
 
-		var pc sidecar.ProviderConfig
-		provider := vaultRoleRegex.FindStringSubmatch(*flagSidecarVaultRole)[2]
+		var provider string
```

Comment:
Think it should be called `sidecarProvider` to disambiguate - since we handle both in the same func.